src/cluster.c

```diff
@@ -5122,7 +5122,7 @@ void addNodeDetailsToShardReply(client *c, clusterNode *node) {
     addReplyBulkCString(c, getPreferredEndpoint(node));
     reply_count++;
 
-    if (node->hostname) {
+    if (sdslen(node->hostname) != 0) {
         addReplyBulkCString(c, "hostname");
         addReplyBulkCString(c, node->hostname);
```

Comment:
if we accept it, should we change this to addReplyBulkCBuffer?
```suggestion
        addReplyBulkCBuffer(c, node->hostname, sdslen(node->hostname));
```

Reply:
Yes we should change it to *CBuffer. Good catch!